src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1021,3 +978,92 @@ EnsureReplicationSettings(Oid relationId, char replicationModel)
 								"factor\" to one%s.", extraHint)));
 	}
 }
+
+
+/*
+ * CopyLocalData copies local data into the shards.
```

Comment:
Added better comments and naming.

> I feel like—in addition to a more descriptive function comment—this function would also benefit from comments breaking it into "sections", i.e. to denote the relevant "steps" in this algorithm, as the bookkeeping variables tend to muck up the vertical space quite badly.

I think the worst offender is the commandList part, so I put that in a separate function. Otherwise, it was kind of tricky to break this up without a lot of duplication (e.g. tupleDescriptor gets used in multiple places), but added some comments to break it up.